apps/test/unit/code-studio/pd/workshop_dashboard/components/workshopFormTest.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {assert, expect} from 'chai';
+import {Factory} from 'rosie';
+import Permission from '@cdo/apps/code-studio/pd/workshop_dashboard/permission';
+import {WorkshopForm} from '@cdo/apps/code-studio/pd/workshop_dashboard/components/workshop_form';
+import '../workshopFactory';
+
+describe('WorkshopForm test', () => {
+  let fakeWorkshop;
+
+  beforeEach(() => {
+    fakeWorkshop = Factory.build('workshop');
+  });
+
+  afterEach(() => {});
+
+  it('renders', () => {
+    const wrapper = shallow(
+      <WorkshopForm
+        permission={new Permission()}
+        facilitatorCourses={[]}
+        workshop={fakeWorkshop}
+        onSaved={() => {}}
+        readOnly={false}
+      />,
+      {context: {router: {}}}
+    );
+
+    const someControl = wrapper.find('FormControl');
+    assert(someControl.exists());
+  });
+
+  it('displays "Location" for a non-virtual workshop', () => {
+    fakeWorkshop.virtual = false;
+    const wrapper = shallow(
+      <WorkshopForm
+        permission={new Permission()}
+        facilitatorCourses={[]}
+        workshop={fakeWorkshop}
+        onSaved={() => {}}
+        readOnly={false}
+      />,
+      {context: {router: {}}}
+    );
+
+    const locationGroup = wrapper.find('#location_name').closest('FormGroup');
```

Comment:
I've been encouraged to import the component itself and use that in Enzyme searches, but I don't really have a strong opinion on the matter...

Reply:
my reasoning is that its easier and reads better to grab by id instead of `.find(FormGroup).childAt(3)`, plus it'd break if the fields got reordered

Reply:
oh but yeah, i can import the component for FormGroup!

Reply:
Interesting -- I do think we do a lot of index type searches, and this does seem more robust! I guess if you can use the component in the that would be nice but not a big deal.